src/core/users/service/users.service.ts

```diff
@@ -0,0 +1,104 @@
+import { Injectable, NotFoundException, Param } from '@nestjs/common';
+import { CreateUserDto } from '../dto/create-user.dto';
+import { UpdateUserDto } from '../dto/update-user.dto';
+import { User } from '../entities/user.entity';
+import { Repository } from 'typeorm';
+import { Profile } from '../../profiles/entities/profile.entity';
+import { InjectRepository } from '@nestjs/typeorm';
+import * as bcryptjs from 'bcryptjs';
+import { JwtPayloadService } from '../../../common/service/jwt.payload.service';
+import { USER_NOT_FOUND } from '../messages';
+export class UsersService {
+  constructor(
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+
+    @InjectRepository(Profile)
+    private readonly profileRepository: Repository<Profile>,
+
+    private readonly jwtPayloadService: JwtPayloadService,
+  ) {}
+
+  async create(createUserDto: CreateUserDto) {
+    const user = await this.userRepository.create(createUserDto);
+
+    createUserDto.password = await bcryptjs.hash(createUserDto.password, 10);
+
+    await this.userRepository.save(user);
+
+    const _user = await this.userRepository.findOne({
+      where: { email: createUserDto.email },
+    });
+
+    const perfil = await this.profileRepository.create({
+      userId: _user.id,
+      description: 'default',
+      mainTitle: 'default',
+      countryResidence: 'default',
```

Comment:
```suggestion
      description: null,
      mainTitle: null,
      countryResidence: null,
```
Por que "default" y no null?

Reply:
ya que no puede ser nulo ese campo